File: Python/fun3.py
def is_prime(n):   # function definition
    if n <= 1:
        return False   # 1 or negative numbers not prime

    for i in range(2, n // 2 + 1):
        if n % i == 0:
            return False   # not prime
    return True    # otherwise prime

# main part
n = int(input("Enter a number: "))

if is_prime(n):   # function call
    print(n, "is Prime")
else:
    print(n, "is Not Prime")

File: Java_Practice/C/fun3.c
#include<stdio.h>
#include<math.h>

int isprime(int n){
    if(n <= 1)
        return 0;

    for(int i = 2; i <= sqrt(n); i++){
        if(n % i == 0)
            return 0; // not prime
    }
    return 1; // prime
}

int main(){
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);

    int result = isprime(n); // function call

    if(result)
        printf("%d is a Prime Number\n", n);
    else
        printf("%d is Not a Prime Number\n", n);

    return 0;
}
